### PR TITLE
Web audio server main and gst player

### PR DESCRIPTION
**Pull request by skywilliamsonl**
Summary: Implement rialto server and gst web audio player functionality.
Type: Feature
Test Plan: Unittests
Jira: RIALTO-2

### CODE REVIEW

**Review by skywojciechowskim — APPROVED** [review]

File: media/server/gstplayer/include/WebAudioPlayerContext.h [hunk 1]
```diff
@@ -21,40 +21,47 @@
...
+     */
+    std::mutex m_writeBufferMutex;
+
+    /**
+     * @brief Write buffer condition variable.
+     */
+    std::condition_variable m_writeBufferCond;
+
+    /**
+     * @brief Write buffer condition variable.
```
Comment (skypetts) [inline media/server/gstplayer/include/WebAudioPlayerContext.h:62]: update comment

File: media/server/gstplayer/source/tasks/webAudio/SetCaps.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    gchar *capsStr = m_gstWrapper->gstCapsToString(caps);
+    std::string strCaps = capsStr;
+    m_glibWrapper->gFree(capsStr);
+
+    RIALTO_SERVER_LOG_DEBUG("caps str: '%s'", strCaps.c_str());
+
+    GstCaps *appsrcCaps = m_gstWrapper->gstAppSrcGetCaps(GST_APP_SRC(m_context.source));
+    if ((!appsrcCaps) || (!m_gstWrapper->gstCapsIsEqual(appsrcCaps, caps)))
+    {
+        RIALTO_SERVER_LOG_MIL("Updating web audio appsrc caps to '%s'", strCaps.c_str());
```
Comment (skypetts) [inline media/server/gstplayer/source/tasks/webAudio/SetCaps.cpp:130]: Should this be _DEBUG?

File: media/server/gstplayer/source/GstWebAudioPlayer.cpp [hunk 3]
```diff
@@ -114,42 +133,247 @@ GstWebAudioPlayer::~GstWebAudioPlayer()
...
-    m_workerThread->join();
-    m_workerThread.reset();
+    resetWorkerThread();
+
+    termWebAudioPipeline();
+}
+
+bool GstWebAudioPlayer::initWebAudioPipeline()
+{
+    m_context.pipeline = m_gstWrapper->gstPipelineNew("webaudiopipeline");
```
Comment (skypetts) [inline media/server/gstplayer/source/GstWebAudioPlayer.cpp:143]: Minor, but would it be helpful to give each pipeline a unique name perhaps based on 'priority'?

**Review by skypetts — APPROVED** [review]

File: media/server/gstplayer/source/tasks/webAudio/SetCaps.cpp (hunk 2 above)
Comment (skywilliamsonl) [reply media/server/gstplayer/source/tasks/webAudio/SetCaps.cpp:130]: We print a Milstone log when setting the caps in MediaPipeline so i did the same here, happy to change it to Debug though

File: media/server/gstplayer/source/GstWebAudioPlayer.cpp (hunk 3 above)
Comment (skywilliamsonl) [reply media/server/gstplayer/source/GstWebAudioPlayer.cpp:143]: Good idea, will do in follow up delivery

File: media/server/gstplayer/include/WebAudioPlayerContext.h (hunk 1 above)
Comment (skywilliamsonl) [reply media/server/gstplayer/include/WebAudioPlayerContext.h:62]: Will fix in follow up delivery.

**Pull request merged**